app/Http/Controllers/PostsController.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Post;
+use App\PostStatus;
+use App\User;
+use Illuminate\Http\Request;
+
+class PostsController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+
+        $posts = Post::all();
+
+        return view('posts.index')->with('posts', $posts);
+    }
+
+    /**
+     * Show the form for creating a new resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function create()
+    {
+        //
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param \Illuminate\Http\Request $request
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function store(Request $request)
+    {
+        //
+    }
+
+    /**
+     * Display the specified resource.
+     *
+     * @param int $id
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function show($id)
+    {
+        $post                      = Post::find($id);
+        $post_attributes           = $post->getAttributes();
+        $post_attributes['author'] = $post->user->name;
+        unset($post_attributes['author_id']);
+        $post_attributes['status'] = $post->status->status;
+        unset($post_attributes['status_id']);
+
+        return view('posts.details')->with('post', $post_attributes);
```

Comment:
I guess this block is not necessary (https://github.com/Bulrati/laravel-roadmap-proj/pull/4/files#diff-f2d947e65c4b6d16491390b642f3d194R56-R61)
You can simply do this by
```
return view('posts.details', ['post' => Post::find($id)]);
```
and get all needed attributes just in view

Reply:
Done.